moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryKnackDataTrackerSync.js

```diff
@@ -33,11 +33,30 @@ const getHttpMethod = knackProjectId => {
  * Function to map the signal IDs from a project object into an array and return the array length.
  * @returns integer
  */
-const countSignalsInProject = project => {
-  const signalIds = project.moped_proj_features.map(
-    feature => feature.location.properties.signal_id
+const getProjectSignals = project => {
+  const allProjectFeatures = project.moped_proj_components
+    .map(projectComponent => {
+      // for each component, extract feature geojson from each project feature (moped_proj_features.feature)
+      // every feature must have a feature.feature jsonb, but we'll nullish chain just to be safe
+      return projectComponent.moped_proj_features.map(
+        feature => feature?.feature
+      );
+    })
+    // then flatten this array of feature arrays
+    .flat()
```

Comment:
TIL. Never would have thought to look for this method. I like the cautionary vibe of it to ensure you have a 1D array.